src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/CriteriaService.java

```diff
@@ -250,10 +251,16 @@ public Question convertRequirementToQuestion(final Requirement r, final String a
                      .selected(o.getSelect() == null? Boolean.FALSE:o.getSelect())
                 .text(o.getText())).collect(Collectors.toList()));
 
+    var description = r.getOcds().getDescription();
+    if (Objects.nonNull(description) && description.contains(END_DATE)) {
+      var agreementDetails = agreementsService.getAgreementDetails(agreementNumber);
+      description = description.replace(END_DATE,
```

Comment:
do we want to `replaceAll(...)` here - just in case the placeholder is used more than once?
Regarding the format - the screenshot has `19 January 2024` - which I think means using `FormatStyle.LONG` ?

Reply:
Made changes